### PR TITLE
Introduce `register_static_xxx!` macros

### PR DESCRIPTION
**Pull request by breeswish**
This can further simplify user code.

### CODE REVIEW

**Comment by breeswish** [comment]
@Hoverbear Friendly ping :)

File: static-metric/examples/register_integration.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#[macro_use]
+extern crate lazy_static;
+#[macro_use]
+extern crate prometheus;
+extern crate prometheus_static_metric;
+
+use prometheus::exponential_buckets;
+use prometheus_static_metric::*;
+
+make_static_metric! {
```
Comment (siddontang) [inline static-metric/examples/register_integration.rs]: may make_static_vec_metric be better?

Comment (breeswish) [reply static-metric/examples/register_integration.rs]: Hmm, a lot of places need to be changed in order to rename (also includes a directory rename), let's do it in another PR.

**Comment by siddontang** [comment]
Ping @Hoverbear

**Pull request merged**